File: c++/program/query.cpp
#include "mahasiswa.cpp"
#include <bits/stdc++.h> //untuk list
using namespace std;
class Query
{
private: // private atribut
    string nama;
    string NIM;
    string prodi;
    string fakultas;
    list<Mahasiswa> db;
    Mahasiswa data;

public:
    Query() /* constructor */
    {
    }
    void addData() // METHOD menambahkan data
    {
        // meminta data yg akan diadd
        cout << "Masukan data\n";
        cout << "Nama :";
        cin >> nama;
        cout << "NIM :";
        cin >> NIM;
        cout << "Prodi :";
        cin >> prodi;
        cout << "fakultas :";
        cin >> fakultas;
        // mengset data yg di input
        data.setNama(nama);
        data.setNIM(NIM);
        data.setProdi(prodi);
        data.setFakultas(fakultas);
        db.insert(db.end(), data); // memasukan data kedlm list
        cout << "------------------------\n";
        cout << "| data telah dimasukan |\n";
        cout << "------------------------\n";
    }
    void deleteData(string hapus) // METHOD menghapus data
    {
        for (auto i = db.begin(); i != db.end(); i++)
        {
            if ((*i).getNIM() == hapus)
            {
                db.erase(i); // hapus data yg nim nya sesuai dengan yg diminta
                break;
            }
        }
        cout << "----------------------\n";
        cout << "| data telah dihapus  |\n";
        cout << "-----------------------\n";
    }
    void updateData(string update) // METHOD UPDATE data
    {
        for (auto i = db.begin(); i != db.end(); i++)
        {
            if ((*i).getNIM() == update)
            {
                cout << "Masukan data yang akan diupdate\n";
                cout << "Nama :";
                cin >> nama;
                cout << "NIM :";
                cin >> NIM;
                cout << "Prodi :";
                cin >> prodi;
                cout << "fakultas :";
                cin >> fakultas;
                (*i).setNama(nama);
                (*i).setNIM(NIM);
                (*i).setProdi(prodi);
                (*i).setFakultas(fakultas);
                break;
            }
        }
        cout << "-----------------------\n";
        cout << "| data telah terupdate |\n";
        cout << "------------------------\n";
    }
    void show() // menampilkan
    {
        for (auto i = db.begin(); i != db.end(); i++)
        {
            cout << (*i).getNama() << " " << (*i).getNIM() << " " << (*i).getProdi() << " " << (*i).getFakultas() << " " << endl; /* code */
        }
    }
    void setNama(string a) // SET NAMA
    {
        nama = a;
    }
    void setNIM(string b) // SET NIM
    {
        NIM = b;
    }
    void setProdi(string c) // SET PRODI
    {
        prodi = c;
    }
    void setFakultas(string d) // SET FAKULTAS
    {
        fakultas = d;
    }
    string getNama() // GET NAMA
    {
        return nama;
    }
    string getNIM() // GET NIM
    {
        return NIM;
    }
    string getProdi() // GET PRODI
    {

        return prodi;
    }
    string getFakultas() // GET FAKULTAS
    {
        return fakultas;
    }
    ~Query()
    {
    }
};

File: PY/PROGRAM/query.py
from mahasiswa import Mahasiswa


class Query:
    def __init__(self, Nama="", Nim="", prodi="", fakultas="", db=[], data=Mahasiswa()):
        self.Nama = Nama
        self.Nim = Nim
        self.prodi = prodi
        self.fakultas = fakultas
        self.db = db
        self.data = data

    def addData(self):  # METHOD menambahkan data
        # meminta data yg akan diadd
        print("Masukan data")
        print("Nama :")
        self.Nama = input()
        print("NIM :")
        self.Nim = input()
        print("Prodi :")
        self.prodi = input()
        print("Fakultas :")
        self.fakultas = input()
        self.data = Mahasiswa(self.Nama, self.Nim, self.prodi, self.fakultas)
        self.db.append(self.data)
        print("------------------------")
        print("| data telah dimasukan |")
        print("------------------------")

    def show(self):  # METHOD MENAMPILKAN data
        for i in self.db:
            print(i.get_Nama()+" "+i.get_Nim()+" " +
                  i.get_prodi()+" "+i.get_fakultas())

    def updateData(self, update):  # METHOD MENGUPDATE data
        for i in self.db:
            if i.get_Nim() == update:
                print("Masukan data untuk update")
                print("Nama :")
                self.Nama = input()
                print("NIM :")
                self.Nim = input()
                print("Prodi :")
                self.prodi = input()
                print("Fakultas :")
                self.fakultas = input()
                i.set_Nama(self.Nama)
                i.set_Nim(self.Nim)
                i.set_prodi(self.prodi)
                i.set_fakultas(self.fakultas)
                print("------------------------")
                print("| data telah terupdate |")
                print("------------------------")

    def deleteData(self, hapus):  # METHOD MENGHAPUS data
        for i in self.db:
            if i.get_Nim() == hapus:
                self.db.remove(i)
                break
        print("------------------------")
        print("| data telah dihapus   |")
        print("------------------------")
